src/apps/vpn/models/recommendedlocationmodel.cpp

```diff
@@ -54,12 +58,21 @@ void RecommendedLocationModel::refreshModel() {
   QList<const ServerCity*> cities = recommendedLocations(DEFAULT_ENTRIES);
   if (m_recommendedCities.length() != cities.length()) {
     beginResetModel();
-    m_recommendedCities.swap(cities);
+    m_recommendedCities.clear();
+    for (auto& city : cities) {
+      // Wrap that pointers into a
+      // QPointer, so that we are notified if they are
```

Comment:
Nit: We aren't really notified -- we can check if the object has been deleted using isNull